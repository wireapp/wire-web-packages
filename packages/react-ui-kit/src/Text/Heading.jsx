/*
 * Wire
 * Copyright (C) 2017 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {COLOR} from '../Identity';
import PropTypes from 'prop-types';
import React from 'react';
import {Text} from './Text';

const H1 = Text.withComponent('h1').extend`
  /* appearance */
  color: ${props => (props.color ? props.color : COLOR.GRAY_LIGHTEN_24)};
  font-size: 48px;
  font-weight: 300;

  /* positioning */
  line-height: 62px;
  margin-bottom: 40px;
  margin-top: 32px;
  min-height: 62px;
`;

const H2 = Text.withComponent('h2').extend`
  /* appearance */
  color: ${props => (props.color ? props.color : COLOR.GRAY_DARKEN_72)};
  font-size: 24px;
  font-weight: 300;
`;

const H3 = Text.withComponent('h3').extend`
  /* appearance */
  color: ${props => (props.color ? props.color : COLOR.GRAY_DARKEN_72)};
  font-size: 18px;
  font-weight: 600;
`;

const H4 = Text.withComponent('h4').extend`
  /* appearance */
  color: ${props => (props.color ? props.color : COLOR.GRAY_DARKEN_72)};
  font-size: 11px;
  font-weight: 700;

  /* positioning */
  margin-bottom: 5px;
  margin-top: 20px;
`;

const Heading = ({level, ...props}) => {
  switch (level) {
    case '2':
      return <H2 {...props} />;
    case '3':
      return <H3 {...props} />;
    case '4':
      return <H4 {...props} />;
    case '1':
    default:
      return <H1 {...props} />;
  }
};

Heading.propTypes = {
  ...Text.propTypes,
  color: PropTypes.string,
  level: PropTypes.oneOf(['1', '2', '3', '4']),
};

Heading.defaultProps = {
  ...Text.defaultProps,
  color: '',
  level: '1',
};

export {Heading, H1, H2, H3, H4};
