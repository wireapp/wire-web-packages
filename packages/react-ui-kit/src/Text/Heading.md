Demo:

```jsx
import React from 'react';
import {Column, Columns, Container, H1, H2, H3, H4, Heading} from '@wireapp/react-ui-kit';

<Container>
  <Columns>
    <Column>Default heading</Column>
    <Column>
      <Heading>Heading default</Heading>
    </Column>
  </Columns>
  <Columns>
    <Column>Heading1</Column>
    <Column>
      <H1>Heading1</H1>
    </Column>
  </Columns>
  <Columns>
    <Column>Heading2</Column>
    <Column>
      <H2>Heading2</H2>
    </Column>
  </Columns>
  <Columns>
    <Column>Heading3</Column>
    <Column>
      <H3>Heading3</H3>
    </Column>
  </Columns>
  <Columns>
    <Column>Heading4</Column>
    <Column>
      <H4>Heading4</H4>
    </Column>
  </Columns>
</Container>;
```
