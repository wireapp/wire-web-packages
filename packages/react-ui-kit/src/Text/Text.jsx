/*
 * Wire
 * Copyright (C) 2017 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import styled, {css} from 'styled-components';
import {COLOR} from '../Identity';
import PropTypes from 'prop-types';

const Text = styled.span`
  /* appearance */
  color: ${props => props.color};
  font-size: ${props => props.fontSize};
  font-weight: ${props => (props.bold ? '600' : 'initial')};
  opacity: ${props => (props.muted ? '0.5' : '1')};
  text-align: ${props => (props.center ? 'center' : 'left')};
  text-transform: ${props => props.textTransform};
  ${props => {
    if (props.noWrap) {
      return css`
        white-space: nowrap;
      `;
    }
  }};
  ${props => {
    if (props.truncate) {
      return css`
        overflow: hidden;
        text-overflow: ellipsis;
      `;
    }
  }};
`;

Text.propTypes = {
  bold: PropTypes.bool,
  center: PropTypes.bool,
  color: PropTypes.string,
  fontSize: PropTypes.string,
  muted: PropTypes.bool,
  noWrap: PropTypes.bool,
  textTransform: PropTypes.oneOf(['lowercase', 'uppercase', 'capitalize', 'inherit', 'none']),
  truncate: PropTypes.bool,
};

Text.defaultProps = {
  bold: false,
  center: false,
  color: COLOR.GRAY_DARKEN_48,
  fontSize: '16px',
  muted: false,
  noWrap: false,
  textTransform: 'none',
  truncate: true,
};

const Bold = Text.withComponent('strong');
Bold.defaultProps = {
  ...Text.defaultProps,
  bold: true,
};

const Small = Text.withComponent('small');
Small.defaultProps = {
  ...Text.defaultProps,
  fontSize: '12px',
};

const Muted = Text.extend``;
Muted.defaultProps = {
  ...Text.defaultProps,
  muted: true,
};

const Uppercase = Text.extend``;
Uppercase.defaultProps = {
  ...Text.defaultProps,
  textTransform: 'uppercase',
};

export {Bold, Muted, Small, Text, Uppercase};
