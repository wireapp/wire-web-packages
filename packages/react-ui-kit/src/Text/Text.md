Demo:

```jsx
import React from 'react';
import {Text} from '@wireapp/react-ui-kit';

<Text>Normal text</Text>;
```
