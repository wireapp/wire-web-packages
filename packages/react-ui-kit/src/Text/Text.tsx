/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import {TextTransformProperty} from 'csstype';
import {COLOR} from '../Identity';
import {filterProps} from '../util';

export interface TextProps<T = HTMLSpanElement> extends React.HTMLProps<T> {
  block?: boolean;
  bold?: boolean;
  center?: boolean;
  color?: string;
  fontSize?: string;
  light?: boolean;
  muted?: boolean;
  noWrap?: boolean;
  textTransform?: TextTransformProperty;
  truncate?: boolean;
}

const filterTextProps = (props: Object) => {
  return filterProps(props, [
    'block',
    'bold',
    'center',
    'color',
    'fontSize',
    'light',
    'muted',
    'noWrap',
    'textTransform',
    'truncate',
  ]);
};

export const textStyles: (props: TextProps) => ObjectInterpolation<undefined> = ({
  block = false,
  bold = false,
  center = false,
  color = COLOR.TEXT,
  fontSize = '16px',
  light = false,
  muted = false,
  noWrap = false,
  textTransform = 'none',
  truncate = false,
}) => ({
  color: color,
  display: block ? 'block' : 'inline',
  fontSize: fontSize,
  fontWeight: bold ? 600 : light ? 200 : 300,
  opacity: muted ? 0.56 : 1,
  overflow: truncate ? 'hidden' : undefined,
  textAlign: center ? 'center' : 'left',
  textOverflow: truncate ? 'ellipsis' : undefined,
  textTransform: textTransform,
  whiteSpace: noWrap ? 'nowrap' : undefined,
});

const Text = (props: TextProps) => <span css={textStyles(props)} {...filterTextProps(props)} />;

const Bold = (props: TextProps) => <Text bold {...props} />;
const Small = (props: TextProps) => <Text fontSize={'12px'} {...props} />;
const Muted = (props: TextProps) => <Text muted {...props} />;
const Uppercase = (props: TextProps) => <Text textTransform={'uppercase'} {...props} />;
const Large = (props: TextProps) => <Text fontSize={'48px'} light {...props} />;

export {Bold, Muted, Small, Text, Uppercase, Large, filterTextProps};
