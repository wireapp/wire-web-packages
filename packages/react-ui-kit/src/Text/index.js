import setGlobalStyles from '../globalStyles';

setGlobalStyles();

export {Heading, H1, H2, H3, H4} from './Heading';
export * from './Line';
export * from './Link';
export * from './Paragraph';
export * from './Text';
