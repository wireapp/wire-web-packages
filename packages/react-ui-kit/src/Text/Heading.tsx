/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import {COLOR} from '../Identity';
import media, {QueryKeys} from '../mediaQueries';
import {TextProps, filterTextProps, textStyles} from './Text';

interface HeadingProps<T = HTMLHeadingElement> extends TextProps<T> {
  level?: string;
}

const h1Styles: (props: HeadingProps) => ObjectInterpolation<undefined> = ({
  block = true,
  color = COLOR.TEXT,
  level = '1',
  noWrap = false,
  textTransform = 'none',
  ...props
}) => ({
  ...textStyles({block, color, noWrap, textTransform, ...props}),
  fontSize: '48px',
  fontWeight: 300,
  lineHeight: '56px',
  marginBottom: '64px',
  marginTop: 0,
  minHeight: '48px',
  [media[QueryKeys.MOBILE]]: {
    fontSize: '40px',
    lineHeight: '48px',
  },
});

const H1 = (props: HeadingProps) => <h1 css={h1Styles(props)} {...filterTextProps(props)} />;

const h2Styles: (props: HeadingProps) => ObjectInterpolation<undefined> = ({
  block = true,
  color = COLOR.TEXT,
  noWrap = false,
  textTransform = 'none',
  ...props
}) => ({
  ...textStyles({block, color, noWrap, textTransform, ...props}),
  fontSize: '24px',
  fontWeight: 700,
  lineHeight: '32px',
  marginBottom: '24px',
  marginTop: '48px',
  [media[QueryKeys.MOBILE]]: {
    fontSize: '20px',
    lineHeight: '28px',
    marginBottom: '20px',
    marginTop: '44px',
  },
});

const H2 = (props: HeadingProps) => <h2 css={h2Styles(props)} {...filterTextProps(props)} />;

const h3Styles: (props: HeadingProps) => ObjectInterpolation<undefined> = ({
  block = true,
  color = COLOR.TEXT,
  noWrap = false,
  textTransform = 'none',
  ...props
}) => ({
  ...textStyles({block, color, noWrap, textTransform, ...props}),
  fontSize: '16px',
  fontWeight: 600,
  marginBottom: '16px',
});

const H3 = (props: HeadingProps) => <h3 css={h3Styles(props)} {...filterTextProps(props)} />;

const h4Styles: (props: HeadingProps) => ObjectInterpolation<undefined> = ({
  block = true,
  color = COLOR.TEXT,
  noWrap = false,
  textTransform = 'none',
  ...props
}) => ({
  ...textStyles({block, color, noWrap, textTransform, ...props}),
  fontSize: '11px',
  fontWeight: 300,
  marginBottom: '5px',
  marginTop: '20px',
});

const H4 = (props: HeadingProps) => <h3 css={h4Styles(props)} {...filterTextProps(props)} />;

const Heading = ({level, ...props}) => {
  switch (level) {
    case '2':
      return <H2 {...props} />;
    case '3':
      return <H3 {...props} />;
    case '4':
      return <H4 {...props} />;
    case '1':
    default:
      return <H1 {...props} />;
  }
};

export {Heading, H1, H2, H3, H4};
