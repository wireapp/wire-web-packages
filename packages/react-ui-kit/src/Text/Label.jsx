/*
 * Wire
 * Copyright (C) 2017 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import PropTypes from 'prop-types';
import React from 'react';
import {Text} from './Text';

const Label = ({component, ...props}) => {
  const StyledLabel = Text.withComponent(component).extend`
    /* positioning */
    width: 100%;
    padding: 24px 0 8px;
  `;
  return <StyledLabel {...props} />;
};

Label.propTypes = {
  ...Text.propTypes,
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
};

Label.defaultProps = {
  ...Text.defaultProps,
  bold: true,
  color: COLOR.LINK,
  component: 'span',
  fontSize: '12px',
};

export {Label};
