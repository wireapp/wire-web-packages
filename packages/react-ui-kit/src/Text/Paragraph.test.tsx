/*
 * Wire
 * Copyright (C) 2019 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import React from 'react';
import {THEME_ID} from '../Layout';
import {matchComponent} from '../test/testUtil';
import {Lead, Paragraph} from '../Text';

describe('"Paragraph"', () => {
  it('renders', () => matchComponent(<Paragraph>Paragraph</Paragraph>));
  it('renders (dark theme)', () => matchComponent(<Paragraph>Paragraph</Paragraph>, THEME_ID.DARK));
});

describe('"Lead"', () => {
  it('renders', () => matchComponent(<Lead>Lead</Lead>));
  it('renders (dark theme)', () => matchComponent(<Lead>Lead</Lead>, THEME_ID.DARK));
});
