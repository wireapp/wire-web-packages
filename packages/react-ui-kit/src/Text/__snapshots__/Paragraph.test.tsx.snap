// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Lead" renders 1`] = `
.emotion-0 {
  color: rgb(52,56,59);
  display: block;
  font-size: 24px;
  font-weight: 300;
  opacity: 1;
  text-align: center;
  text-transform: none;
  margin-bottom: 56px;
  margin-top: 0;
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 20px;
  }
}

<p
  className="emotion-0"
>
  Lead
</p>
`;

exports[`"Paragraph" renders 1`] = `
.emotion-0 {
  color: rgb(52,56,59);
  display: block;
  font-size: 16px;
  font-weight: 300;
  opacity: 1;
  text-align: left;
  text-transform: none;
  margin-bottom: 16px;
  margin-top: 0;
}

<p
  className="emotion-0"
>
  Paragraph
</p>
`;
